src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1195,22 +1192,56 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
+                        hasErrors = true;
+                    }
+                    break;
+            }
 
-            // PROTOTYPE(ngafter): TODO: check that the operator exists for that type (e.g. error for string < "")
-            BinaryOperatorKind kind = TokenKindToBinaryOperatorKind(node.OperatorToken.Kind());
-            return new BoundRelationalPattern(node, kind, value, constantValueOpt ?? ConstantValue.Bad, inputType, hasErrors);
-        }
+            if (constantValueOpt is null) hasErrors = true;
+            return new BoundRelationalPattern(node, operation | opType, value, constantValueOpt ?? ConstantValue.Bad, inputType, hasErrors);
 
-        private BinaryOperatorKind TokenKindToBinaryOperatorKind(SyntaxKind kind) => kind switch
-        {
-            SyntaxKind.LessThanEqualsToken => BinaryOperatorKind.LessThanOrEqual,
-            SyntaxKind.LessThanToken => BinaryOperatorKind.LessThan,
-            SyntaxKind.GreaterThanToken => BinaryOperatorKind.GreaterThan,
-            SyntaxKind.GreaterThanEqualsToken => BinaryOperatorKind.GreaterThanOrEqual,
-            // The following occur in error recovery scenarios
-            SyntaxKind.ExclamationEqualsToken => BinaryOperatorKind.NotEqual,
-            _ => BinaryOperatorKind.Equal,
-        };
+            static BinaryOperatorKind TokenKindToBinaryOperatorKind(SyntaxKind kind) => kind switch
+            {
+                SyntaxKind.LessThanEqualsToken => BinaryOperatorKind.LessThanOrEqual,
+                SyntaxKind.LessThanToken => BinaryOperatorKind.LessThan,
+                SyntaxKind.GreaterThanToken => BinaryOperatorKind.GreaterThan,
+                SyntaxKind.GreaterThanEqualsToken => BinaryOperatorKind.GreaterThanOrEqual,
+                // The following occurs in error recovery scenarios
+                _ => BinaryOperatorKind.Equal,
+            };
+
+            static BinaryOperatorKind BinaryOperatorType(SpecialType specialType) => specialType switch
+            {
+                SpecialType.System_Single => BinaryOperatorKind.Float,
+                SpecialType.System_Double => BinaryOperatorKind.Double,
+                SpecialType.System_Char => BinaryOperatorKind.Char,
+                SpecialType.System_SByte => BinaryOperatorKind.Int, // operands are converted to int
+                SpecialType.System_Byte => BinaryOperatorKind.Int, // operands are converted to int
+                SpecialType.System_UInt16 => BinaryOperatorKind.Int, // operands are converted to int
+                SpecialType.System_Int16 => BinaryOperatorKind.Int, // operands are converted to int
+                SpecialType.System_Int32 => BinaryOperatorKind.Int,
+                SpecialType.System_UInt32 => BinaryOperatorKind.UInt,
+                SpecialType.System_Int64 => BinaryOperatorKind.Long,
+                SpecialType.System_UInt64 => BinaryOperatorKind.ULong,
+                _ => 0,
```

Comment:
>0 [](start = 21, length = 1)

nit: `None`

Reply:
No such thing.  Using `BinaryOperatorKind.Error` (which is 0)

---
In reply to: [381005281](https://github.com/dotnet/roslyn/pull/41503#discussion_r381005281) [](ancestors = 381005281)